ambry-router/src/main/java/com/github/ambry/router/GetBlobOperation.java

```diff
@@ -1107,29 +1107,40 @@ private void processGetBlobResponse(GetRequestInfo getRequestInfo, GetResponse g
       }
     }
 
+    /**
+     * Process response if it was rejected due to quota compliance.
+     * @param correlationId correlation id of the request.
+     * @param replicaId {@link ReplicaId} of the request.
+     */
+    private void processQuotaRejectedResponse(int correlationId, ReplicaId replicaId) {
+      logger.trace("GetBlobRequest with response correlationId {} rejected because it exceeded quota", correlationId);
+      onErrorResponse(replicaId, new RouterException("QuotaExceeded", RouterErrorCode.TooManyRequests), false);
+      checkAndMaybeComplete();
```

Comment:
checking if the operation tracker is failed due to quota?

Reply:
Since `onErrorResponse` sets the operation exception, we actually don't need to check fail on quota anywhere. I have updated other places to reflect that. Thanks for pointing this out.